controllers/machinehealthcheck_targets.go

```diff
@@ -170,7 +170,7 @@ func (r *MachineHealthCheckReconciler) getTargetsFromMHC(clusterClient client.Re
 
 	targets := []healthCheckTarget{}
 	for k := range machines {
-		patchHelper, err := patch.NewHelper(&machines[k], r.Client)
+		patchHelper, err := patch.NewHelper(&machines[k], r.Client, patch.WithStatusObservedGeneration{})
```

Comment:
We should not update observed generation for resources that we do not manage, only the Machine controller should update the observed generation for Machines.
```suggestion
		patchHelper, err := patch.NewHelper(&machines[k], r.Client)
```